source/dub/compilers/ldc.d

```diff
@@ -322,4 +322,9 @@ config    /etc/ldc2.conf (x86_64-pc-linux-gnu)
 				    || arg.startsWith("-mtriple=");
 		}
 	}
+
+	bool isWindowsCOFF(in BuildPlatform platform)
+	{
+		return platform.isWindows && platform.architecture.canFind("x86", "x86_64");
```

Comment:
What about ARM bed windows? This should just `return platform.isWindows;`, no?

Reply:
ARM idk about. I have no experience there, no way to test or any idea what the architecture string would be.

No its MSVC linker or compatible linker, PE-COFF only. So no Optlink.

Reply:
LDC doesn't ship ARM builds, has dub even been tested on ARM Windows?

Reply:
> So no Optlink.

Correct, but LDC doesn't support that anyway IIRC, or is DUB still capable of passing that to LDC  and still need to be filtered out?

> LDC doesn't ship ARM builds,

Cross-compilation.

> has dub even been tested on ARM Windows?

No idea.

> No its MSVC linker or compatible linker, PE-COFF only.

More silly questions: what does LLD (e.g. `-link-internally`) fall into then? 
does it change its command line to be MSVC compatible on windows or is there a separate driver for it?

Reply:
Gotcha, yeah it can probably be shrunk down a bit, will do.

LLD is MSVC link compatible (its in changelog that it is).

> $ lld-link /?
> OVERVIEW: LLVM Linker
> 
> USAGE: C:\Tools\D\dmd_2.102.0\windows\bin64\lld-link.exe [options] file...
> 
> OPTIONS:
>   /align:<value>          Section alignment
>   /aligncomm:<value>      Set common symbol alignment
>   /allowbind:no           Disable DLL binding
>   /allowbind              Enable DLL binding (default)
>   /allowisolation:no      Disable DLL isolation
>   /allowisolation         Enable DLL isolation (default)
>   /alternatename:<value>  Define weak alias
>   /appcontainer:no        Image can run outside an app container (default)
>   /appcontainer           Image can only be run in an app container
>   /base:<value>           Base address of the program
>   /Brepro                 Use a hash of the executable as the PE header timestamp
>   --color-diagnostics=<value>
>                           Use colors in diagnostics; one of 'always', 'never', 'auto'
>   --color-diagnostics     Use colors in diagnostics
>   /debug:<value>          Embed a symbol table in the image with option
>   /debugtype:<value>      Debug Info Options
>   /debug                  Embed a symbol table in the image
>   /def:<value>            Use module-definition file
>   /defaultlib:<value>     Add the library to the list of input files
>   /delayload:<value>      Delay loaded DLL name
>   /demangle:no            Do not demangle symbols in output
>   /demangle               Demangle symbols in output (default)
>   /dll                    Create a DLL
>   /driver:<value>         Generate a Windows NT Kernel Mode Driver
>   /dynamicbase:no         Disable ASLR (default when /fixed)
>   /dynamicbase            Enable ASLR (default unless /fixed)
>   /entry:<value>          Name of entry point symbol
>   /errorlimit:<value>     Maximum number of errors to emit before stopping (0 = no limit)
>   /export:<value>         Export a function
>   /failifmismatch:<value>
>   /filealign:<value>      Section alignment in the output file
>   /fixed:no               Enable base relocations (default)
>   /fixed                  Disable base relocations
>   /force:multipleres      Allow multiply defined resources when creating executables
>   /force:multiple         Allow multiply defined symbols when creating executables
>   /force:unresolved       Allow undefined symbols when creating executables
>   /force                  Allow undefined and multiply defined symbols when creating executables
>   /functionpadmin:<value> Prepares an image for hotpatching
>   /guard:<value>          Control flow guard
>   /heap:<value>           Size of the heap
>   /highentropyva:no       Disable 64-bit ASLR
>   /highentropyva          Enable 64-bit ASLR (default on 64-bit)
>   /ignore:<value>         Specify warning codes to ignore
>   /implib:<value>         Import library name
>   /include:<value>        Force symbol to be added to symbol table as undefined one
>   /includeoptional:<value>
>                           Add symbol as undefined, but allow it to remain undefined
>   /incremental:no         Overwrite import library even if contents are unchanged
>   /incremental            Keep original import library if contents are unchanged
>   /integritycheck:no      No effect (default)
>   /integritycheck         Set FORCE_INTEGRITY bit in PE header
>   /largeaddressaware:no   Disable large addresses (default on 32-bit)
>   /largeaddressaware      Enable large addresses (default on 64-bit)
>   /libpath:<value>        Additional library search path
>   /lib                    Act like lib.exe; must be first argument if present
>   /linkrepro:<value>      Dump linker invocation and input files for debugging
>   /lldltocache:<value>    Path to ThinLTO cached object file directory
>   /lldltocachepolicy:<value>
>                           Pruning policy for the ThinLTO cache
>   /lldsavetemps           Save temporary files instead of deleting them
>   /machine:<value>        Specify target platform
>   /manifest:<value>       NO disables manifest output; EMBED[,ID=#] embeds manifest as resource in the image
>   /manifestdependency:<value>
>                           Attributes for <dependency> element in manifest file; implies /manifest
>   /manifestfile:<value>   Manifest output path, with /manifest
>   /manifestinput:<value>  Additional manifest inputs; only valid with /manifest:embed
>   /manifestuac:<value>    User access control
>   /manifest               Create .manifest file
>   /merge:<value>          Combine sections
>   /mllvm:<value>          Options to pass to LLVM
>   /natvis:<value>         Path to natvis file to embed in the PDB
>   /no-color-diagnostics   Do not use colors in diagnostics
>   /nodefaultlib:<value>   Remove a default library
>   /nodefaultlib           Remove all default libraries
>   /noentry                Don't add reference to DllMainCRTStartup; only valid with /dll
>   /nxcompat:no            Disable data execution provention
>   /nxcompat               Enable data execution prevention (default)
>   /opt:<value>            Control optimizations
>   /order:<value>          Put functions in order
>   /out:<value>            Path to file to write output
>   /pdb:<value>            PDB file path
>   /pdbaltpath:<value>     PDB file path to embed in the image
>   /pdbsourcepath:<value>  Base path used to make relative source file path absolute in PDB
>   --rsp-quoting=<value>   Quoting style for response files, 'windows' (default) or 'posix'
>   /safeseh:no             Don't produce an image with Safe Exception Handler
>   /safeseh                Produce an image with Safe Exception Handler (only for x86)
>   /section:<value>        Specify section attributes
>   /stack:<value>          Size of the stack
>   /stub:<value>           Specify DOS stub file
>   /subsystem:<value>      Specify subsystem
>   /swaprun:cd             Make loader run output binary from swap instead of from CD
>   /swaprun:net            Make loader run output binary from swap instead of from network
>   /swaprun:<value>        Comma-separated list of 'cd' or 'net'
>   /thinlto-emit-imports-files
>                           Emit .imports files with -thinlto-index-only
>   /thinlto-index-only:<value>
>                           -thinlto-index-only and also write native module names to file
>   /thinlto-index-only     Instead of linking, emit ThinLTO index files
>   /thinlto-object-suffix-replace:<value>
>                           'old;new' replace old suffix with new suffix in ThinLTO index
>   /thinlto-prefix-replace:<value>
>                           'old;new' replace old prefix with new prefix in ThinLTO outputs
>   /threads:no             Do not run the linker multi-threaded
>   /threads                Run the linker multi-threaded (default)
>   /timestamp:<value>      Specify the PE header timestamp
>   /tsaware:no             Create non-Terminal Server aware executable
>   /tsaware                Create Terminal Server aware executable (default)
>   /version:<value>        Specify a version number in the PE header
>   --version               Print version information
>   /wholearchive:<value>   Include all object files from this archive
>   /WX:no                  Don't treat warnings as errors
>   /WX                     Treat warnings as errors